adapters/pulsepoint_test.go

```diff
@@ -0,0 +1,193 @@
+ */
+func TestPulsePointAdapterNames(t *testing.T) {
+	adapter := NewPulsePointAdapter(DefaultHTTPAdapterConfig, "http://localhost/bid", "http://localhost")
+	VerifyStringValue(adapter.Name(), "PulsePoint", t)
+	VerifyStringValue(adapter.FamilyName(), "pulsepoint", t)
+}
+
+/**
+ * Verifies the user sync parameters.
+ */
+func TestPulsePointUserSyncInfo(t *testing.T) {
+	adapter := NewPulsePointAdapter(DefaultHTTPAdapterConfig, "http://localhost/bid", "http://localhost")
+	VerifyStringValue(adapter.GetUsersyncInfo().Type, "redirect", t)
+	VerifyStringValue(adapter.GetUsersyncInfo().URL, "https://bh.contextweb.com/rtset?pid=-1&ev=1&rurl=http%3A%2F%2Flocalhost%2Fsetuid%3Fbidder%3Dpulsepoint%26uid%3D%25%25VGUID%25%25", t)
+}
+
+/**
+ * Test required parameters not sent
+ */
+func TestPulsePointRequiredBidParameters(t *testing.T) {
+	adapter := NewPulsePointAdapter(DefaultHTTPAdapterConfig, "http://localhost/bid", "http://localhost")
+	ctx := context.TODO()
+	req := SampleRequest()
+	bidder := SampleBidder()
+	// remove "ct" param and verify error message.
+	bidder.AdUnits[0].Params = json.RawMessage("{\"cp\": 2001, \"cf\": \"728X90\"}")
+	_, errTag := adapter.Call(ctx, &req, &bidder)
+	VerifyStringValue(errTag.Error(), "Missing TagId param ct", t)
+	// remove "cp" param and verify error message.
+	bidder.AdUnits[0].Params = json.RawMessage("{\"ct\": 1001, \"cf\": \"728X90\"}")
+	_, errPub := adapter.Call(ctx, &req, &bidder)
+	VerifyStringValue(errPub.Error(), "Missing PublisherId param cp", t)
+	// remove "cf" param and verify error message.
+	bidder.AdUnits[0].Params = json.RawMessage("{\"cp\": 2001, \"ct\": 1001}")
+	_, errSize := adapter.Call(ctx, &req, &bidder)
+	VerifyStringValue(errSize.Error(), "Missing AdSize param cf", t)
+}
+
+/**
+ * Verify the openrtb request sent to Pulsepoint endpoint.
+ * Ensure the ct, cp, cf params are transformed and sent alright.
+ */
+func TestPulsePointOpenRTBRequest(t *testing.T) {
+	var ortbRequest openrtb.BidRequest
+	server := httptest.NewServer(
+		http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+			// Send 204 (no-bid)
+			decoder := json.NewDecoder(req.Body)
+			decoder.Decode(&ortbRequest)
+			defer req.Body.Close()
+			w.WriteHeader(204)
+		}),
+	)
+	defer server.Close()
+	ctx := context.TODO()
+	req := SampleRequest()
+	bidder := SampleBidder()
+	adapter := NewPulsePointAdapter(DefaultHTTPAdapterConfig, server.URL, "http://localhost")
+	adapter.Call(ctx, &req, &bidder)
+	VerifyStringValue(ortbRequest.Imp[0].TagID, "1001", t)
+	VerifyStringValue(ortbRequest.Site.Publisher.ID, "2001", t)
+	VerifyIntValue(int(ortbRequest.Imp[0].Banner.W), 728, t)
+	VerifyIntValue(int(ortbRequest.Imp[0].Banner.H), 90, t)
+}
+
+/**
+ * Verify bidding behavior.
+ */
+func TestPulsePointBiddingBehavior(t *testing.T) {
+	// setup server endpoint to return bid.
+	server := httptest.NewServer(
+		http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+			resp := SampleBidResponse()
```

Comment:
This function should closely match the functionality of the PulsePoint openrtb endpoint. I'm assuming it does some validation of inbound requests - please implement that here (see facebook_test.go for example based on their published spec). If there's a link to a public endpoint spec would be great to include up top so that people can validate.